### DISCUSSION TITLE
@midwayjs/mock里的src得改成调用tsconfig.json里的rootDir

### DISCUSSION THREAD

**Discussion by muyetuge** [midwayjs/midway#3683] [Q&A] [answered]
### Describe the problem（描述问题）

@midwayjs/mock里的src得改成调用tsconfig.json里的rootDir  不然修改rootdir后测试会找不到框架

### Midway Versions（Midway 版本）

>> Start to check your midway component version...

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(not installed)
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.15.0)
✓ @midwayjs/bull(not installed)
✓ @midwayjs/bull-board(not installed)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.15.0)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(not installed)
✓ @midwayjs/decorator(3.15.0)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(not installed)
✓ @midwayjs/info(3.15.1)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.15.0)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(not installed)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(not installed)
✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.15.1)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.15.1)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment]
是指单测？

**Reply by muyetuge** [reply]
"rootDir":"src/server" 改完rootDir路径运行pnpm test
位置 node_modules\@midwayjs\mock\dist\creator.js
`options.baseDir = (0, path_1.join)(appDir, 'src');`

**Reply by czy88840616** [reply]
是用了 @midwayjs/cli 吧，可以换 mwtsc 了，mwtsc 里已经处理了。

**Reply by TsMask** [reply]
v4版本后全面切换mwtsc吗？

**Reply by czy88840616** [reply]
会，现在新的脚手架也都已经换过了。

**Reply by TsMask** [reply]
希望v4后以node20起步，ems依赖都能用上，装饰器ts5.5可以兼容旧的吗？

**Reply by czy88840616** [reply]
不影响吧，ts5 现在也能用，都是在项目依赖中的，mwtsc 也是使用项目的 tsc。

**Reply by TsMask** [reply]
好的，期待v4新特性

**Reply by muyetuge** [reply]
不是呀 在esm项目里测的  
是这个mocha测试
 `"test": "cross-env NODE_ENV=unittest mocha",`

**Reply by czy88840616** [reply, marked as answer]
测试可以改 baseDir 的

```ts
const app = await createApp({
  baseDir: join(__dirname, 'src')
})
